### ISSUE TITLE
[Bug] Countdown: Date picker in settings shifts 1 day back

### ISSUE THREAD

**Issue by Yixuan-Wang** [KieronQuinn/SmartspacerPlugins#28]
### Select the Plugin you are reporting for

Countdown

### Description

<img src="https://github.com/KieronQuinn/SmartspacerPlugins/assets/21151119/e5d5a980-0296-4dd7-b9f5-abd7694c71a0" width="45%" alt="date picker">
<img src="https://github.com/KieronQuinn/SmartspacerPlugins/assets/21151119/c276efff-504a-41e5-b514-88ecddb5654e" width="45%" alt="settings">

When you select a day in the `Countdown to` option, the date always shifts 1 day backwards. If you open the date picker, do nothing and press "Ok", the date will continue to shift 1 day back. 

Currently I'm in UTC-6 timezone, which might be relevant to the bug.

I think the problem should be around here during Instant conversion: 
https://github.com/KieronQuinn/SmartspacerPlugins/blob/6983c3859c318fcee585c50757c02e70abb39d25/countdown/src/main/java/com/kieronquinn/app/smartspacer/plugin/countdown/ui/screens/configuration/ConfigurationViewModel.kt#L80C5

**Comment by Yixuan-Wang** [comment]
After consulting ChatGPT, I think the fix it provided should be correct. The epoch returned from the [DatePicker](https://developer.android.com/reference/com/google/android/material/datepicker/MaterialDatePicker) is based on UTC, not the device's local timezone? You are in UTC+0 so this bug cannot be reproduced.

```kotlin
val instant = Instant.ofEpochMilli(date)
val localDate = instant.atZone(ZoneOffset.UTC).toLocalDate()
// Format the date as needed
val formattedDate = localDate.format(DateTimeFormatter.ISO_LOCAL_DATE)

updateComplicationData {
    it.copy(endDate = formattedDate)
}
```

**Comment by KieronQuinn** [comment]
Fixed in countdown v1.0.1, ChatGPT was right for once in that the timezone is indeed meant to be in UTC
